util/execdetails/execdetails.go

```diff
@@ -457,6 +457,14 @@ func (e *RuntimeStats) SetRowNum(rowNum int64) {
 	atomic.StoreInt64(&e.rows, rowNum)
 }
 
+// ClearConcurrencyInfo clears the concurrency information.
+// We must call clearConcurrencyInfo first when we need to call SetConcurrencyInfo.
```

Comment:
If so, how about call this function in `SetConcurrencyInfo` and remove in other place?

Reply:
In [`Hash Agg`](https://github.com/pingcap/tidb/blob/d62fcd408e19b24e6d5704d1f66fc652ce386ed5/executor/aggregate.go#L246-L248 ) , we have two concurrency information need to set. So I think it's better to keep two functions. Or we can change `SetConcurrencyInfo` to get more concurrency information and set them all at once. What's your opinion?  @crazycs520

Reply:
> Or we can change SetConcurrencyInfo to get more concurrency information and set them all at once.

I prefer this way. The current implementation requires each executor to clear their concurrency information after execution, which is hardly to achieve by different contributors.